gammapy/datasets/map.py

```diff
@@ -1961,6 +1961,17 @@ def to_region_map_dataset(self, region, name=None):
         name = make_name(name)
         kwargs = {"gti": self.gti, "name": name, "meta_table": self.meta_table}
 
+        if self.mask and not self.mask.geom.is_region:
+            region_mask = self.mask.geom.region_mask(region)
+            for mask_image, region_mask_image in zip(
+                self.mask.iter_by_image(), region_mask.iter_by_image()
+            ):
+                if len(np.unique(mask_image.data[region_mask_image.data])) > 1:
```

Comment:
why not checking `np.all(mask_image.data[region_mask_image.data])`

Reply:
A priori, masks are only booleans. If not I would even simply raise an error here. What is the meaning of the projection in this case becomes very unclear in my opinion.

Reply:
mask_image combines all energy with a or condition but we need to check each energy bin in case the mask change with energy.
and I use np.unique instead of np.all because all False in one energy bin is also acceptable.